### PR TITLE
Add open lineage support for MergeOperator

### PR DESCRIPTION
**Pull request by sunank200**
# Description
## What is the current behavior?

Open Lineage support for `MergeOperator` is missing.

closes: #900 

## What is the new behavior?

- Add open Lineage support for `MergeOperator`
- Add the test for the same

## Does this introduce a breaking change?
No

### Checklist
- [x] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: python-sdk/src/astro/sql/operators/merge.py [hunk 1]
```diff
@@ -73,8 +83,62 @@ def execute(self, context: Context) -> BaseTable:
...
+                    ),
+                    "output_stats": OutputStatisticsOutputDatasetFacet(rowCount=self.target_table.row_count),
+                },
+            )
+        ]
+
+        run_facets: dict[str, BaseFacet] = {}
+
+        merge_query = task_instance.xcom_pull(task_ids=task_instance.task_id, key="merge_query")
+        job_facets: dict[str, BaseFacet] = {"sql": SqlJobFacet(query=str(merge_query))}
```
Comment (pankajastro) [inline python-sdk/src/astro/sql/operators/merge.py:136]: @sunank200 do we need typecast `str` since we are pushing str in xcom?

**Review by pankajastro — APPROVED** [review]

**Comment by pankajastro** [comment]
LGTM

File: python-sdk/src/astro/sql/operators/merge.py (hunk 1 above)
Comment (sunank200) [reply python-sdk/src/astro/sql/operators/merge.py:136]: I can remove this

**Comment by pankajastro** [comment]
@sunank200 to fix the test look like you will have to pass context having ti here https://github.com/astronomer/astro-sdk/blob/main/python-sdk/tests/sql/operators/test_merge.py#L297 something like https://github.com/astronomer/astro-sdk/blob/main/python-sdk/tests/sql/operators/test_append.py#L81

**Comment by sunank200** [comment]
> Add context to the task in test

yup https://github.com/astronomer/astro-sdk/pull/1068/commits/224e4de4d8fed82f8b3f9f38426c205d9c5efa57

File: python-sdk/src/astro/sql/operators/merge.py [hunk 2]
```diff
@@ -73,8 +83,62 @@ def execute(self, context: Context) -> BaseTable:
             target_conflict_columns=self.target_conflict_columns,
             source_to_target_columns_map=self.columns,
         )
+        context["ti"].xcom_push(key="merge_query", value=str(db.sql))
```
Comment (uranusjr) [inline python-sdk/src/astro/sql/operators/merge.py:86]: Why is this additional `str` needed?

Comment (pankajastro) [reply python-sdk/src/astro/sql/operators/merge.py:86]: I think because sqlalchemy insert statement is not serializable

Comment (uranusjr) [reply python-sdk/src/astro/sql/operators/merge.py:86]: Ah OK that makes sense, I thought `sql` is the raw string.

Comment (sunank200) [reply python-sdk/src/astro/sql/operators/merge.py:86]: @uranusjr we wanted to typecast this to string before pushing as per discussion: https://astronomer.slack.com/archives/C03868KGF2Q/p1665754738297099

**Pull request merged**